### PR TITLE
Battle Algorithm related fixes from battle 7

### PR DESCRIPTION
**Pull request by fmatthew5876**
Battle 7 commits related to battle algorithm calculations

### CODE REVIEW

**Comment by fmatthew5876** [comment]
This one is tested and ready. I have nothing more to add here.

**Comment by carstene1ns** [comment]
jenkins: test this please (newlib check).

File: src/game_battlealgorithm.cpp [hunk 1]
```diff
@@ -931,7 +931,7 @@ bool Game_BattleAlgorithm::Normal::Execute() {
 		}
 		if (GetTarget()->IsDefending()) {
 			if (GetTarget()->HasStrongDefense()) {
-				effect /= 3;
+				effect /= 4;
```
Comment (Ghabry) [inline src/game_battlealgorithm.cpp]: didn't cherry mention somewhere that the value is 3?

File: src/game_battlealgorithm.cpp [hunk 2]
```diff
@@ -1104,16 +1103,17 @@ bool Game_BattleAlgorithm::Skill::Execute() {
...
-			}
-
-			effect +=
+			int effect = skill.power +
 				source->GetAtk() * skill.physical_rate / 20 +
 				source->GetSpi() * skill.magical_rate / 40;
-			effect *= mul;
+
+			// Determined via testing, the heal is always at least 50%
+			effect *= std::max(0.5f, GetAttributeMultiplier(skill.attribute_effects));
```
Comment (Ghabry) [inline src/game_battlealgorithm.cpp]: here is the 50%-thing again

Comment (fmatthew5876) [reply src/game_battlealgorithm.cpp]: Fixed

Comment (Ghabry) [reply src/game_battlealgorithm.cpp]: iirc, I also saw this 50%-thing a few lines below again.

File: src/game_battlealgorithm.cpp (hunk 1 above)
Comment (fmatthew5876) [reply src/game_battlealgorithm.cpp]: The value is 4.

I made physical skill with 0 str, 0 mag, 0 var, base damage 20.

When it hits Alex:
* Normal 20 dmg
* Defending 10 dmg
* Defending with Super 5 dmg

**Comment by Ghabry** [comment]
@fmatthew5876 merge conflict :/

**Comment by fmatthew5876** [comment]
rebased to master

**Pull request merged**